src/main/java/useCases/generator/RevisedPropertyGenerator.java

```diff
@@ -0,0 +1,54 @@
+package useCases.generator;
+
+import entity.Property;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+/**
+ * It is a generator generating the property blocks
+ */
+public class RevisedPropertyGenerator {
+
+    static final String DELIMITER = ",";
+
+    /**
+     * Create properties based on the information file
+     */
+
+    public static ArrayList<Property> generateProperties(String fileName){
+
+        ArrayList<Property> propertyList = new ArrayList<>();
+
+        try (BufferedReader br = new BufferedReader(new FileReader(fileName))){
+            String line;
+            ArrayList<Integer> priceList = new ArrayList<>(Arrays.asList(0, 0, 0, 0, 0, 0));
+            ArrayList<Integer> taxList = new ArrayList<>(Arrays.asList(0, 0, 0, 0, 0, 0));
+
+            while ((line = br.readLine()) != null) {
+                String[] values = line.split(DELIMITER);
+
+                // generate the priceList, length 6 with last element being 0.
+                for (int i = 0; i < 5; i++) {
+                    priceList.set(i, Integer.parseInt(values[i + 2]));
```

Comment:
I think this is great, the original one only supports up to 3 different levels, which is probably not how Monopoly works